### PR TITLE
Added DocBlocks to methods that have an array as return type.

### LINKED ISSUE #97: Blocks on protected keywords protection where it doesn't make sense.
```fpp
MyNamespace {
    data Condition =  New | AsNew | Good | Reasonable | Moderate deriving(Enum);
}
```

This gives an error because the `New` keyword is in there, but in my opinion this should be allowed.

Also (as in singular of 'returns'):

```fpp
MyNamespace\Return {

}
```

```
Parse Error: Syntax error, unexpected 'Return', expecting 'T_STRING' at line 1 on file 'domain.fpp'
```

### PR DESCRIPTION
**Pull request by paales**
Fixes #97

### CODE REVIEW

File: src/helpers.php [hunk 1]
```diff
@@ -139,14 +139,37 @@ function buildArgumentReturnType(Argument $argument, Definition $definition): st
...
+
 function buildScalarConstructor(Definition $definition): string
 {
     return "new {$definition->name()}(\$value)";
 }
 
 function buildArgumentConstructor(Argument $argument, Definition $definition, DefinitionCollection $collection): string
 {
-    if ($argument->isScalartypeHint() || null === $argument->type()) {
+    if ($argument->isScalarTypeHint() || null === $argument->type()) {
```
Comment (paales) [inline src/helpers.php]: @prolic I did change this, not sure if this has something to do with the builds failing. My PHPStorm was complaining here about a typo.

**Comment by paales** [comment]
@prolic Some tests are failing, not totally sure if it is because of my code, can't make sense of the test failing.

File: src/builder/buildAccessors.php [hunk 2]
```diff
@@ -45,8 +46,9 @@ function buildAccessors(Definition $definition, ?Constructor $constructor, Defin
                     $methodBody = buildMethodBodyFromPayload($argument, $definition, $collection, true);
                 }
 
+                $accessors .= buildDocBlockReturnType($argument);
                 $accessors .= <<<CODE
-    public function {$argument->name()}()$returnType
+    public function {$argument->type()}()$returnType
```
Comment (prolic) [inline src/builder/buildAccessors.php]: why change this?

Comment (prolic) [reply src/builder/buildAccessors.php]: I think that's why travis tests fail

**Comment by paales** [comment]
Will check if I've got the time.

**Comment by prolic** [comment]
@paales check this example:
```
namespace Foo;

data Test = Test {string[] $foo};
data Bar = Bar {string $foo, Test[] $tests};
```

this will generate this method:
```
    public function withTests(array $tests): Bar
    {
        return new self($this->foo, $tests);
    }
```

and
```
    public function withFoo(array $foo): Test
    {
        return new self(...$foo);
    }
```

there is the new docblock missing.

**Comment by prolic** [comment]
I added docblock for setters and constructors

**Pull request merged**